lib/utils.js

```diff
@@ -86,21 +86,36 @@ function patchProperties(obj, properties) {
 
 function patchEventTargetMethods(obj) {
   var addDelegate = obj.addEventListener;
-  obj.addEventListener = function (eventName, fn) {
-    fn._bound = fn._bound || {};
-    arguments[1] = fn._bound[eventName] = zone.bind(fn);
+  obj.addEventListener = function (eventName, handler) {
+    var fn;
+    
+    if (handler.handleEvent) {
+      // Have to pass in 'handler' reference as an argument here, otherwise it gets clobbered in
+      // IE9 by the arguments[1] assignment at end of this function.
+      fn = (function(handler) {
+        return function() {
+          handler.handleEvent.apply(handler, arguments);
+        };
+      })(handler);
+    } else {
+      fn = handler;
+    }
+
+    handler._fn = fn;
+    handler._bound = handler._bound || {};
+    arguments[1] = handler._bound[eventName] = zone.bind(fn);
     return addDelegate.apply(this, arguments);
   };
 
   var removeDelegate = obj.removeEventListener;
-  obj.removeEventListener = function (eventName, fn) {
+  obj.removeEventListener = function (eventName, handler) {
     if(arguments[1]._bound && arguments[1]._bound[eventName]) {
```

Comment:
Agreed. line 114 will have to stay the same, though, unless we want to 'manually' pass each argument to the delegate.